File: frontend/src/types/auditLog.ts
export type AuditActorType = "USER" | "CLIENT" | "SYSTEM";

export interface AuditLogEntry {
  id: string;
  actor_type: AuditActorType;
  actor_user_id: string | null;
  actor_client_id: string | null;
  action: string;
  resource_type: string;
  resource_id: string | null;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
}

File: backend/app/schemas/audit_log.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.audit_log import AuditActorType


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    actor_type: AuditActorType
    actor_user_id: uuid.UUID | None
    actor_client_id: uuid.UUID | None
    action: str
    resource_type: str
    resource_id: uuid.UUID | None
    metadata_json: dict | None
    created_at: datetime
